README
NAME

    Data::Object::Role::Dumpable

ABSTRACT

    Dumpable Role for Perl 5

SYNOPSIS

      package Example;
    
      use Moo;
    
      with 'Data::Object::Role::Dumpable';
    
      package main;
    
      my $example = Example->new;
    
      # $example->dump

DESCRIPTION

    This package provides methods for dumping the object and underlying
    value.

METHODS

    This package implements the following methods:

 dump

      dump() : Str

    The dump method returns a string representation of the underlying data.

    dump example #1

        # given: synopsis
      
        my $dumped = $example->dump;

 pretty_dump

      pretty_dump() : Str

    The pretty_dump method returns a string representation of the
    underlying data that is human-readable and useful for debugging.

    pretty_dump example #1

        # given: synopsis
      
        my $dumped = $example->pretty_dump;

 pretty_print

      pretty_print(Any @args) : Int

    The pretty_print method prints a stringified human-readable
    representation of the underlying data.

    pretty_print example #1

        # given: synopsis
      
        my $printed = $example->pretty_print;

    pretty_print example #2

        # given: synopsis
      
        my $printed = $example->pretty_print({1..4});

 pretty_say

      pretty_say(Any @args) : Int

    The pretty_say method prints a stringified human-readable
    representation of the underlying data, with a trailing newline.

    pretty_say example #1

        # given: synopsis
      
        my $printed = $example->pretty_say;

    pretty_say example #2

        # given: synopsis
      
        my $printed = $example->pretty_say({1..4});

 print

      print(Any @args) : Int

    The print method prints a stringified representation of the underlying
    data.

    print example #1

        # given: synopsis
      
        my $printed = $example->print;

    print example #2

        # given: synopsis
      
        my $printed = $example->print({1..4});

 say

      say(Any @args) : Int

    The say method prints a stringified representation of the underlying
    data, with a trailing newline.

    say example #1

        # given: synopsis
      
        my $printed = $example->say;

    say example #2

        # given: synopsis
      
        my $printed = $example->say({1..4});

AUTHOR

    Al Newkirk, awncorp@cpan.org

LICENSE

    Copyright (C) 2011-2019, Al Newkirk, et al.

    This is free software; you can redistribute it and/or modify it under
    the terms of the The Apache License, Version 2.0, as elucidated in the
    "license file"
    <https://github.com/iamalnewkirk/foobar/blob/master/LICENSE>.

PROJECT

    Wiki <https://github.com/iamalnewkirk/foobar/wiki>

    Project <https://github.com/iamalnewkirk/foobar>

    Initiatives <https://github.com/iamalnewkirk/foobar/projects>

    Milestones <https://github.com/iamalnewkirk/foobar/milestones>

    Contributing
    <https://github.com/iamalnewkirk/foobar/blob/master/CONTRIBUTE.md>

    Issues <https://github.com/iamalnewkirk/foobar/issues>



CHANGES
Changelog for Data-Object-Role-Dumpable

2.01 2020-02-17T04:50:36
 - Added release updates
 - Re-releasing to the CPAN

2.00 2020-02-15T02:40:01
 - Built release version 2.00
 - Added release updates
 - Initial implementation
 - Initial commit


lib/Data/Object/Role/Dumpable.pm
package Data::Object::Role::Dumpable;

use 5.014;

use strict;
use warnings;
use routines;

use Moo::Role;

# VERSION

# METHODS

method dump() {
  require Data::Dumper;
  require Data::Object::Utility;

  no warnings 'once';

  local $Data::Dumper::Indent = 0;
  local $Data::Dumper::Purity = 0;
  local $Data::Dumper::Quotekeys = 0;
  local $Data::Dumper::Deepcopy = 1;
  local $Data::Dumper::Deparse = 1;
  local $Data::Dumper::Sortkeys = 1;
  local $Data::Dumper::Terse = 1;
  local $Data::Dumper::Useqq = 1;

  my $data = Data::Dumper::Dumper($self);

  $data =~ s/^"|"$//g;

  return $data;
}

method pretty_dump() {
  require Data::Dumper;
  require Data::Object::Utility;

  no warnings 'once';

  local $Data::Dumper::Indent = 2;
  local $Data::Dumper::Trailingcomma = 0;
  local $Data::Dumper::Purity = 0;
  local $Data::Dumper::Pad = '';
  local $Data::Dumper::Varname = 'VAR';
  local $Data::Dumper::Useqq = 0;
  local $Data::Dumper::Terse = 1;
  local $Data::Dumper::Freezer = '';
  local $Data::Dumper::Toaster = '';
  local $Data::Dumper::Deepcopy = 1;
  local $Data::Dumper::Quotekeys = 0;
  local $Data::Dumper::Bless = 'bless';
  local $Data::Dumper::Pair = ' => ';
  local $Data::Dumper::Maxdepth = 0;
  local $Data::Dumper::Maxrecurse = 1000;
  local $Data::Dumper::Useperl = 0;
  local $Data::Dumper::Sortkeys = 1;
  local $Data::Dumper::Deparse = 1;
  local $Data::Dumper::Sparseseen = 0;

  my $data = Data::Dumper::Dumper($self);

  $data =~ s/^'|'$//g;

  chomp $data;

  return $data;
}

method pretty_print(@args) {

  return $self->printer(map &pretty_dump($_), $self, @args);
}

method pretty_say(@args) {

  return $self->printer(map +(&pretty_dump($_), "\n"), $self, @args);
}

method print(@args) {

  return $self->printer(map &dump($_), $self, @args);
}

method printer(@args) {

  return CORE::print(@args);
}

method say(@args) {

  return $self->printer(map +(&dump($_), "\n"), $self, @args);
}

1;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;

use 5.014;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Dumpable Role for Perl 5",
  "AUTHOR" => "Al Newkirk <awncorp\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Data-Object-Role-Dumpable",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014",
  "NAME" => "Data::Object::Role::Dumpable",
  "PREREQ_PM" => {
    "Moo" => 0,
    "routines" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Moo" => 0,
    "Test::Auto" => "0.05",
    "routines" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "2.00",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Moo" => 0,
  "Test::Auto" => "0.05",
  "routines" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
